app/models/embed/purl/resource_xml_deserializer.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Embed
+  class Purl
+    class ResourceXmlDeserializer
+      # @param [String] druid the identifier of the resource
+      # @param [Nokogiri::XML::Element] resource
+      # @param [Dor::RightsAuth] rights
+      def initialize(druid, resource, rights)
+        @druid = druid
+        @resource = resource
+        @rights = rights
+      end
+
+      def deserialize
+        description = @resource.xpath('./label').text
+        files = @resource.xpath('./file').map do |file|
+          FileXmlDeserializer.new(@druid, description, file, @rights).deserialize
+        end
```

Comment:
These could be private methods... but I'd say that's a relatively low-value refactoring.